sql/core/src/main/scala/org/apache/spark/sql/streaming/GroupState.scala

```diff
@@ -205,92 +205,127 @@ trait GroupState[S] extends LogicalGroupState[S] {
   /** Get the state value as a scala Option. */
   def getOption: Option[S]
 
-  /**
-   * Update the value of the state. Note that `null` is not a valid value, and it throws
-   * IllegalArgumentException.
-   */
-  @throws[IllegalArgumentException]("when updating with null")
+  /** Update the value of the state. */
   def update(newState: S): Unit
 
   /** Remove this state. */
   def remove(): Unit
 
   /**
    * Whether the function has been called because the key has timed out.
-   * @note This can return true only when timeouts are enabled in `[map/flatmap]GroupsWithStates`.
+   * @note This can return true only when timeouts are enabled in `[map/flatmap]GroupsWithState`.
    */
   def hasTimedOut: Boolean
 
+
```

Comment:
nit: extra line

Reply:
thats intentional. visually makes it easier to read.